File: backend/src/main/java/com/prueba/backend/service/BenefitService.java
package com.prueba.backend.service;

import com.prueba.backend.dto.RequestDTO.BenefitRequestDTO;
import com.prueba.backend.dto.ResponseDTO.BenefitResponseDTO;
import com.prueba.backend.entity.Benefit;

import java.util.List;

/**
 * Contrato de servicios para la gestión de beneficios laborales.
 *
 * <p>Define las operaciones CRUD disponibles para la entidad {@code Benefit}.
 * La implementación concreta se encuentra en {@link com.prueba.backend.service.impl.BenefitServiceImpl}.
 */
public interface BenefitService {

    /**
     * Obtiene todos los beneficios registrados en el sistema.
     *
     * @return lista de {@link BenefitResponseDTO} con todos los beneficios
     */
    List<BenefitResponseDTO> findAll();

    /**
     * Obtiene todos los beneficios asociados a un empleado específico.
     *
     * @param employeeId identificador del empleado
     * @return lista de {@link BenefitResponseDTO} de ese empleado (puede estar vacía)
     */
    List<BenefitResponseDTO> findAllByIdEmployee(Long employeeId);

    /**
     * Crea y persiste un nuevo beneficio para un empleado.
     *
     * @param dto datos de entrada del beneficio
     * @return {@link BenefitResponseDTO} con los datos del beneficio creado
     * @throws com.prueba.backend.exception.ResourceNotFoundException si el empleado indicado no existe
     */
    BenefitResponseDTO save(BenefitRequestDTO dto);

    /**
     * Actualiza los datos de un beneficio existente.
     * Permite cambiar también el empleado al que está asignado.
     *
     * @param id  identificador del beneficio a actualizar
     * @param dto nuevos datos del beneficio
     * @return {@link BenefitResponseDTO} con los datos actualizados
     * @throws com.prueba.backend.exception.ResourceNotFoundException si el beneficio o el empleado no existen
     */
    BenefitResponseDTO update(Long id, BenefitRequestDTO dto);

    /**
     * Elimina un beneficio por su identificador.
     *
     * @param id identificador del beneficio a eliminar
     * @throws com.prueba.backend.exception.ResourceNotFoundException si el beneficio no existe
     */
    void delete(Long id);
}

File: frontend/src/services/benefitService.js
import api from "../api/axios";

/**
 * Servicio de acceso a la API REST para la gestión de beneficios laborales.
 * Encapsula todas las peticiones HTTP relacionadas con el recurso `/benefits`.
 */
export default {
  /**
   * Obtiene todos los beneficios de un empleado específico.
   * @param {number|string} idEmployee - Identificador del empleado
   * @returns {Promise} respuesta de Axios con `data` como arreglo de beneficios
   */
  getBenefitsByEmployee(idEmployee) {
    return api.get(`/benefits/employee/${idEmployee}`);
  },

  /**
   * Crea un nuevo beneficio.
   * @param {Object} benefit - Datos del beneficio (benefitName, amount, employeeId)
   * @returns {Promise} respuesta de Axios con el beneficio creado en `data`
   */
  createBenefit(benefit) {
    return api.post("/benefits", benefit);
  },

  /**
   * Actualiza un beneficio existente.
   * @param {number|string} id - Identificador del beneficio a actualizar
   * @param {Object} benefit - Nuevos datos del beneficio
   * @returns {Promise} respuesta de Axios con el beneficio actualizado en `data`
   */
  updateBenefit(id, benefit) {
    return api.put(`/benefits/${id}`, benefit);
  },

  /**
   * Elimina un beneficio por su ID.
   * @param {number|string} id - Identificador del beneficio a eliminar
   * @returns {Promise} respuesta de Axios (204 No Content)
   */
  deleteBenefit(id) {
    return api.delete(`/benefits/${id}`);
  },
};
